lib/mix/tasks.ex

```diff
@@ -80,13 +85,29 @@ defmodule Mix.Tasks.Coveralls do
       end
     end)
 
+    sub_dir_set? = "" != common_options[:subdir] |> to_string
+    root_dir_set? = "" != common_options[:rootdir] |> to_string
```

Comment:
it's not necessary to pipe `to_string`, look:

```elixir
iex(1)> "" != "a"
true
iex(2)> "" != "a" |> to_string
true
```

Reply:
It is necessary since `nil` can come in. I already added unit test for the case. Though I'll the code more explicit.
```
iex(28)> "" == nil |> to_string
true
```

Reply:
But if you change the other you call the condition? Look:
```elixir
iex(1)> nil == ""
false
iex(2)> "" == nil
false
iex(3)> "" == nil |> to_string
true
iex(4)> nil == "" |> to_string
false
```

Reply:
I'm not quite sure what you mean, but if `common_options[:rootdir]` can be `nil`, I think `to_string/1` is necessary to compare with an empty string `""`. By the way I modified this part to make it clear that, it also checks `nil`, so hopefully the intention is more explicit now.

Reply:
Nice, I like this approach and you do your intentions more clearly